File: 1112/r2lm-master/r2lm.m
function r2lmvalue = r2lm( measuredy, estimatedy )
% r^2 based on the latest measured y-values
%
% Calculate r^2 based on the latest measured y-values
% measuredy and estimatedy must be vectors.
%
% --- input ---
% measuredy : measured y-values ( m x 1, m is the number of samples)
% estimated : estimated y-values ( m x 1 )
%
% --- output ---
% r2lmvalue : r2 based on the latest measured y-values
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

r2lmvalue = 1 - sum( (measuredy-estimatedy).^2 ) ./ sum( (measuredy(1:end-1,:)-measuredy(2:end,:)).^2 );

end

File: 1112/r2lm-master/r2lm.py
# r^2 based on the latest measured y-values
import numpy as np


# Calculate r^2 based on the latest measured y-values
# measured_y and estimated_y must be vectors.
def r2lm(measured_y, estimated_y):
    measured_y = np.array(measured_y).flatten()
    estimated_y = np.array(estimated_y).flatten()
    return float(1 - sum((measured_y - estimated_y) ** 2) / sum((measured_y[1:] - measured_y[:-1]) ** 2))
